enclave/src/channel_storage.rs

```diff
@@ -27,20 +27,22 @@ use std::sync::{
 };
 
 lazy_static! {
-    static ref GLOBAL_CHANNEL_STORAGE: Arc<SgxMutex<ChannelStorage>> =
-        Arc::new(SgxMutex::new(ChannelStorage::default()));
+    static ref GLOBAL_CHANNEL_STORAGE: Arc<SgxMutex<Option<ChannelStorage>>> =
+        Arc::new(SgxMutex::new(None));
```

Comment:
Is the Mutex still necessary? Sender can be cloned with every call, so a lock is actually unnecassary, or am I overlooking something?

Reply:
Sorry for being picky about this, but locking takes time, so if we don't need it, we should'nt use it.

Reply:
Sender<ChannelType> cannot be shared between threads safely, so the mutex is still necessary there

Reply:
Why not? According to: https://doc.rust-lang.org/std/sync/mpsc/struct.Sender.html

>  `pub struct Sender<T> { /* fields omitted */ }`
> 
> The sending-half of Rust’s asynchronous channel type. This half can only be owned by one thread, but it can be cloned to send to other threads.

Reply:
I might be doing something wrong here, but I'm trying to remove the mutex and lazy_static won't let me use Sender
```
29 | / lazy_static! {
30 | |     static ref GLOBAL_CHANNEL_STORAGE: Arc<Option<ChannelStorage>> = Arc::new(None);
31 | | }
   | |_^ `std::sync::mpsc::Sender<ChannelType>` cannot be shared between threads safely
   | 
  ::: /root/.cargo/registry/src/github.com-1ecc6299db9ec823/lazy_static-1.4.0/src/core_lazy.rs:12:20
   |
12 |   pub struct Lazy<T: Sync>(Once<T>);
   |                      ---- required by this bound in `lazy_static::lazy::Lazy`
```

Reply:
Check out: https://stackoverflow.com/questions/40384274/rust-mpscsender-cannot-be-shared-between-threads
respective:
https://teaclave.apache.org/api-docs/crates-enclave/sgx_tstd/sync/mpsc/struct.SyncSender.html

Reply:
Oh, but then without a Mutex we lose mutability.

Reply:
Not sure I can follow this comment.